delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphIndexingStream.scala

```diff
@@ -0,0 +1,93 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.indexing
+
+import cats.syntax.functor._
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.ClasspathResourceUtils
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.client.BlazegraphClient
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.model.BlazegraphView.IndexingBlazegraphView
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.model.BlazegraphViewsConfig
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.RemoteContextResolution
+import ch.epfl.bluebrain.nexus.delta.sdk.ProgressesStatistics.ProgressesCache
+import ch.epfl.bluebrain.nexus.delta.sdk.model.BaseUri
+import ch.epfl.bluebrain.nexus.delta.sdk.views.indexing.IndexingStream.ProgressStrategy
+import ch.epfl.bluebrain.nexus.delta.sdk.views.indexing.{IndexingSource, IndexingStream}
+import ch.epfl.bluebrain.nexus.delta.sdk.views.model.ViewIndex
+import ch.epfl.bluebrain.nexus.delta.sourcing.projections.ProjectionId.ViewProjectionId
+import ch.epfl.bluebrain.nexus.delta.sourcing.projections.ProjectionProgress.NoProgress
+import ch.epfl.bluebrain.nexus.delta.sourcing.projections.ProjectionStream.{ChunkStreamOps, SimpleStreamOps}
+import ch.epfl.bluebrain.nexus.delta.sourcing.projections.{Projection, ProjectionProgress}
+import fs2.Stream
+import monix.bio.{IO, Task}
+import monix.execution.Scheduler
+
+/**
+  * Defines how to build a stream for an IndexingBlazegraphView
+  */
+final class BlazegraphIndexingStream(
+    client: BlazegraphClient,
+    indexingSource: IndexingSource,
+    cache: ProgressesCache,
+    config: BlazegraphViewsConfig,
+    projection: Projection[Unit]
+)(implicit cr: RemoteContextResolution, baseUri: BaseUri, sc: Scheduler)
+    extends IndexingStream[IndexingBlazegraphView] {
+
+  implicit private val cl: ClassLoader = getClass.getClassLoader
+
+  override def apply(
+      view: ViewIndex[IndexingBlazegraphView],
+      strategy: IndexingStream.ProgressStrategy
+  ): Stream[Task, Unit] =
+    Stream
+      .eval {
+        // Evaluates strategy and set/get the appropriate progress
+        createNamespace(view.index) >> handleProgress(strategy, view.projectionId)
+      }
+      .flatMap { progress =>
+        indexingSource(view.projectRef, progress.offset, view.resourceTag)
+          .evalMapValue(BlazegraphIndexingStreamEntry.fromEventExchange(_))
+          .evalMapFilterValue {
```

Comment:
Having it combined in one stream evaluation it does not make it easier to follow, but actually more difficult (since you have to map to a Some, even though that part of the evaluation never discards).

I agree that it is more difficult to follow because there is part of the logic in `fromEventExchange` and part in the `IndexingStream` but that's due to the fact that this `fromEventExchange` can be then reused in `CompositeIndexingStream` and `BlazegraphIndexingStream`